app/controllers/VoxelyticsController.scala

```diff
@@ -159,87 +158,78 @@ class VoxelyticsController @Inject()(
 
   def storeWorkflowEvents(workflowHash: String, runName: String): Action[List[WorkflowEvent]] =
     sil.SecuredAction.async(validateJson[List[WorkflowEvent]]) { implicit request =>
-      def createWorkflowEvent(runId: ObjectId, event: WorkflowEvent): Fox[Unit] =
-        event match {
-          case ev: RunStateChangeEvent => voxelyticsDAO.upsertRunStateChangeEvent(runId, ev)
+      def createWorkflowEvent(runId: ObjectId, events: List[WorkflowEvent]): Fox[Unit] =
+        if (events.nonEmpty) {
+          events.head match {
```

Comment:
I’d prefer functional style here
```suggestion
        events.headOption.map { event =>
          event match { … }
        }.getOrElse(Fox.successful())
```